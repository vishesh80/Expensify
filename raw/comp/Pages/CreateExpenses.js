import React from 'react';
import ExpenseForm from '../ExpenseForm';
import { connect } from 'react-redux';
import { add } from '../../modules/actions';

const c = p => (

<div>
    <h1>Create account</h1>
        <ExpenseForm submit={expense => { p.dispatch(add(expense)); p.history.push('/')}}/>
</div>);

export default connect()(c);