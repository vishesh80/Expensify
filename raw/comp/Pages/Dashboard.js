import React from 'react';
import {connect} from 'react-redux';

import ExpenseItem from '../ExpenseItem';
import Filters from '../Filters';
import filter from '../../modules/filter';

export let d = (props) => (

<div>

    <h1>Dashboard</h1>
    <Filters/>
    <ul>
        {filter(props.expenses,props.filters).map(i => <ExpenseItem key={i.id} {...i}/>)}
    </ul>
</div>

);

function stateToProps(state)
{
    return {expenses : state.expenses,
            filters : state.filters
        };
}

export default connect(stateToProps)(d);