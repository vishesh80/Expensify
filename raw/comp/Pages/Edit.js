import React from 'react';
import ExpenseForm from '../ExpenseForm';
import { connect } from 'react-redux';
import { edit } from '../../modules/actions';

const Edit = p =>  (
    <div>
        <h1>Edit Expense Item</h1>
        <ExpenseForm 
                initialState = {p.expenses.filter(e => p.match.params.id === e.id)[0]} 
                submit = {changes => {changes.id = p.match.params.id; p.dispatch(edit(changes)); p.history.push('/');}}
        />
    </div>);
    


function stateToProps(state) {

    return {
        expenses: state.expenses,
    };
}

export default connect(stateToProps)(Edit);