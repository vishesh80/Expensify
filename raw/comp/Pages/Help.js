import React from 'react';


const h = () => (<h1>Help</h1>);

export default h;