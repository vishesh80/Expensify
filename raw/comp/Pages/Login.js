import React from 'react';
import {auth , googleAuth} from '../../modules/firebase';

const l = () => (
  <div className="background">
    <div className="modal">
      <h1>Expensify</h1>
      <p>Its time to get your expenses under control.</p>
      <button onClick={(e) => auth.signInWithPopup(googleAuth)}>Login with Google</button>
    </div>
  </div>
);

export default l;