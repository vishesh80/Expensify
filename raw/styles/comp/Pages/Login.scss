.background
{
    background-image: url('/images/bg.jpg');
    width: 100%;
    height: 100%;
    background-size: cover;
    display: flex;
    justify-content: space-around;
    align-items: center;
}

.modal
{
    padding: 1.5em;
    display: inline-block;
    background-color: rgba(255, 255, 255, 0.781);
    width: 360px;
    text-align: center;
    margin: auto auto;

    button
    {
        background-color: rgb(32, 166, 255);
        font-size: 1.1rem;
        padding: .4em .8em;
        border: none;
        color: white;
    }

    h1
    {
        margin-top: 0;
    }
}