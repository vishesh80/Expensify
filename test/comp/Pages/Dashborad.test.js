import { shallow } from "enzyme";
import moment from "moment";
import React from "react";
import { d as D } from "../../../raw/comp/Pages/Dashboard";
import eToJ from "enzyme-to-json";

const props = {
  filters: {
    text: "",
    startDate: null,
    endDate: null,
    sortBy: "amount",
  },

  expenses: [
    {
      description: "eg",
      amount: 100,
      createdAt: moment(0).add(1, "day"),
      id: "eg",
      dispatch: function (r) {
        return r;
      }
    }
  ]
};

test("Dashboard component test", () => {
  
  let wrapper = shallow(<D {...props} />);
  expect(eToJ(wrapper)).toMatchSnapshot();
});
