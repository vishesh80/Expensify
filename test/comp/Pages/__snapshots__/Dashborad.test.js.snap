// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component test 1`] = `
<div>
  <h1>
    Dashboard
  </h1>
  <React.Memo />
  <ul>
    <React.Memo
      amount={100}
      createdAt={"1970-01-02T00:00:00.000Z"}
      description="eg"
      dispatch={[Function]}
      id="eg"
      key="eg"
    />
  </ul>
</div>
`;
